Contributing

nigthknight opened the pull request:
To enable Contributions by the community we need to define some standards. 
This is a first WIP specification for Contributing Guidelines.

nigthknight committed:
Add first version of CONTRIBUTING.md

nigthknight committed:
[WIP] Update CONTRIBUTING

nigthknight committed:
[WIP] Add Headline for CLA

nigthknight commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+
+## Code Style
+
+## Contributor License Agreement (CLA)
```
Researching other projects I noticed almost all the bigger projects use some kind of CLA. Do we need that?

Weltraumschaf replied:
Not sure. Until we know that we should leave it out. I made a ticket in our backlog.

nigthknight commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+
+## Code of Conduct
+
+## Code Style
```
We should define some basic style guidelines. For Go, we could take a look at https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md

nigthknight commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+
+## Code Review
+
+## Code of Conduct
```
I'm not sure if this is necessary at this early stage

J12934 replied:
It's always better to have a coc before anything happens where you would need to reference a coc.

There are some standard ones availible: https://www.contributor-covenant.org/version/1/4/code-of-conduct/

nigthknight commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+
+NOTE: Make sure you don't include `@mentions` or `fixes` keywords in your git commit messages. These should be included in the PR body instead.
+
+## Code Review
```
We should define some review guidelines. (rel. Code Style)

Weltraumschaf replied:
I would prefer code style definitons for IDE checked in because nobody wants to make this by hand nowadays.

nigthknight commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+5. merge only after pull request review
+6. deploy immediately after review
+
+## Working with Forks and Pull Requests
```
A Pull Request template could be useful.

Weltraumschaf replied:
Indeed. I made a ticket in the backlog.

nigthknight committed:
Adding init CODE_OF_CONDUCT

nigthknight commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Should we refer to the Slack channel?

Weltraumschaf replied:
IMHO email is a better option because it has lower entry hurdle. We should think about "securecodebox [at] iteratec [dot] com".

nigthknight replied:
I opened a ticket in the backlog

nigthknight committed:
Add reference to CODE_OF_CONDUCT

nigthknight committed:
Adding Table of Contents

nigthknight committed:
Fix Syntax for Table of Contents

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+[homepage]: https://www.contributor-covenant.org
+
+For answers to common questions about this code of conduct, see the FAQ at
+https://www.contributor-covenant.org/faq. Translations are available at
```
Plain link in Markdown should be enclsoed by `<` and `>`.

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage],
+version 2.0, available at
+https://www.contributor-covenant.org/version/2/0/code_of_conduct.html.
```
Plain link in Markdown should be enclsoed by < and >.

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
```
The dot should be escaped in Markdown like `### 1\. Correction`.

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
```
The dot should be escaped in Markdown like `### 1\. Correction`.

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+like social media. Violating these terms may lead to a temporary or
+permanent ban.
+
+### 3. Temporary Ban
```
The dot should be escaped in Markdown like `### 1\. Correction`.

Weltraumschaf commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,130 @@
+with those enforcing the Code of Conduct, is allowed during this period.
+Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
```
The dot should be escaped in Markdown like `### 1\. Correction`.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+- Feature branch
+- Main branch (previously called master)
+
+The `feature` branches are used to develop new features as well as fixes.
```
IMHO you should emphasize named things with `_` like _feature_, because the backticks are for preformatted text like code.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+
+TL;DR
+
+1. anything in the main branch is deployable
```
Here you should also emphasize _main_ to clarify that it means the _main_ introduced above.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+
+## Working with Forks and Pull Requests
+
+Our current continuous integration workflow makes it very hard to work on own forks of the SecureCodeBox because CI tests cannot be executed outside of our repository. We are aware of that problem and are working on a solution.
```
_secureCodeBox_ starts lower case andshould be emphasized.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+
+Our current continuous integration workflow makes it very hard to work on own forks of the SecureCodeBox because CI tests cannot be executed outside of our repository. We are aware of that problem and are working on a solution.
+
+Generally the SCB project follows the standard [GitHub Pull request process](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).
```
We should not use _SCB_ abbreviated because a newbie may not know what it stand for.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+
+Generally the SCB project follows the standard [GitHub Pull request process](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).
+
+## Working with Issues / How to Contribute in Issues
```
There should be no white space around the slash.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+
+## Working with Issues / How to Contribute in Issues
+
+For every task taking more than 1h, opening an issue is required.
```
I suggest the phrase: It is mandatory to open an issue, if take longer than one hour.

Weltraumschaf commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,80 @@
+## Working with Issues / How to Contribute in Issues
+
+For every task taking more than 1h, opening an issue is required.
+Before opening an Issue please verify there is no existing one covering your issue.
```
I suggest: Before you open an issue please verify ...

nigthknight committed:
Remove CLA section

We are not sure if we need an agreement like this
Until then we will remove this section

nigthknight committed:
Enclose plain links in angle brackets

nigthknight committed:
Escape special char in headings

nigthknight committed:
Fix markdown syntax

nigthknight committed:
Change wording

nigthknight committed:
Remove dead link

nigthknight committed:
Merge branch 'master' into contributing

nigthknight committed:
Remove Heading

nigthknight committed:
Add Email Address for CoC Complaints

J12934 commented on ressources.md:
```diff
@@ -0,0 +1,15 @@
+# Ressources for Contributing
+
+## Contributing
+
+- [Kubernetes Contributing](https://github.com/kubernetes/community/blob/master/contributors/guide/contributing.md#learn-about-sigs)
+- [Kubernetes Pull-requests](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md)
+- [Nodejs Contributing](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md)
+
+## Code of Conduct
+
+- [Kubernetes Code of Conduct](https://github.com/kubernetes/community/blob/master/governance.md#code-of-conduct)
+
+## Developer's Certificate of Origin? (Nodejs)
+
+## Rebase over Merge? (Nodejs)
```
This all looks still to be WIP?

Is this intended to be changed later? If so, we should probably open up a ticket for it.

nigthknight replied:
Good catch! This was never meant to be committed.

nigthknight committed:
Remove unused file from Repo
